internal/manager/task_stash_box_tag.go

```diff
@@ -258,6 +293,260 @@ func (t *StashBoxPerformerTagTask) stashBoxPerformerTag(ctx context.Context) {
+		if txnErr != nil {
+			logger.Warnf("error while executing read transaction: %v", err)
+		}
+		if remoteID != "" {
+			studio, err = client.FindStashBoxStudio(ctx, remoteID)
+		}
+	} else {
+		var name string
+		if t.name != nil {
+			name = *t.name
+		} else {
+			name = t.studio.Name
+		}
+		studio, err = client.FindStashBoxStudio(ctx, name)
+	}
+
+	if err != nil {
+		logger.Errorf("Error fetching studio data from stash-box: %s", err.Error())
+		return
+	}
+
+	excluded := map[string]bool{}
+	for _, field := range t.excluded_fields {
+		excluded[field] = true
+	}
+
+	// studio will have a value if pulling from Stash-box by Stash ID or name was successful
+	if studio != nil {
+		var dbInput models.StudioDBInput
+		var err error
+
+		// Refreshing an existing studio
+		if t.studio != nil {
+			if studio.Parent != nil && t.create_parent {
+				if studio.Parent.StoredID == nil {
+					// The parent needs to be created
+					dbInput.ParentCreate, err = studioFromScrapedStudio(ctx, studio.Parent, t.box.Endpoint, excluded)
+					if err != nil {
+						logger.Errorf("Failed to make parent studio from scraped studio %s: %s", studio.Parent.Name, err.Error())
+						return
+					}
+				} else {
+					// The parent studio matched an existing one and the user has chosen in the UI to link and/or update it
+					dbInput.ParentUpdate, err = studioPartialFromScrapedStudio(ctx, studio.Parent, studio.Parent.StoredID, t.box.Endpoint, excluded)
+					if err != nil {
+						logger.Errorf("Failed to make parent studio partial from scraped studio %s: %s", studio.Parent.Name, err.Error())
+						return
+					}
+				}
+			}
+
+			dbInput.StudioUpdate, err = studioPartialFromScrapedStudio(ctx, studio, studio.StoredID, t.box.Endpoint, excluded)
+			if err != nil {
+				logger.Errorf("Failed to make studio partial from scraped studio %s: %s", studio.Name, err.Error())
+				return
+			}
+
+			// Start the transaction and update the studio
+			err = txn.WithTxn(ctx, instance.Repository, func(ctx context.Context) error {
+				qb := instance.Repository.Studio
+
+				if err := ValidateModifyStudio(ctx, *dbInput.StudioUpdate, qb); err != nil {
+					return err
+				}
+
+				_, err = qb.UpdatePartial(ctx, dbInput)
+				return err
+			})
+			if err != nil {
+				logger.Errorf("Failed to execute partial update of studio %s: %s", studio.Name, err.Error())
+			} else {
+				logger.Infof("Updated studio %s", studio.Name)
+			}
+
+			//TODO: This wasn't previously part of batch performer updates, but it probably should be for both perfomer and studio?
+			/*
+				if runParentCreateHook {
+					r.hookExecutor.ExecutePostHooks(ctx, *updatedStudio.ParentID, plugin.StudioCreatePost, input, nil)
+				} else if runParentUpdateHook {
+					r.hookExecutor.ExecutePostHooks(ctx, *updatedStudio.ParentID, plugin.StudioUpdatePost, input, parentTranslator.getFields())
+				}
+				r.hookExecutor.ExecutePostHooks(ctx, updatedStudio.ID, plugin.StudioUpdatePost, input, translator.getFields())
+			*/
```

Comment:
Wondering if the hooks should be run during batch operations.
If so, they will need to refactored to be available to this package somehow.